### PR TITLE
pin python version for cmake workflows

### PR DESCRIPTION
**Pull request by pmeier**
On macOS, we are downloading Miniconda with 

https://github.com/pytorch/vision/blob/46b7e27122deb011a3ff09720da9c3787757f7eb/.circleci/config.yml#L986

Until late Dec 2022 this included Python 3.9. From then on, the included Python version was 3.10 and our workflows started [failing due to some unsatisfiable requirements](https://app.circleci.com/pipelines/github/pytorch/vision/22481/workflows/d5e77548-16ba-42c7-8bf0-fd7674a3a76c/jobs/1803108).

Closes https://github.com/pytorch/vision/issues/7073

### CODE REVIEW

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -986,7 +986,7 @@ jobs:
             curl -o conda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
             sh conda.sh -b
             source $HOME/miniconda3/bin/activate
-            conda install -yq conda-build cmake
+            conda install -yq conda-build cmake python=3.9
```
Comment (NicolasHug) [inline .circleci/config.yml.in:989]: For consistency with the other cmake jobs, I think this should be 

```suggestion
            conda install -yq conda-build cmake python=PYTHON_VERSION
```

which would require an extra section above like:

```
    environment:
      PYTHON_VERSION: << parameters.python_version >>
```

**Review by NicolasHug — APPROVED** [review]
Thanks for the PR @pmeier !

LGTM although LMK what you think on my comment below. I'll approve regardless so feel free to merge once you're happy with the solution

File: .circleci/config.yml.in (hunk 1 above)
Comment (pmeier) [reply .circleci/config.yml.in:989]: Good point!

**Comment by pmeier** [comment]
On Windows we are simply using the Python that is installed on the image, which is 3.7: https://app.circleci.com/pipelines/github/pytorch/vision/22531/workflows/dc22a930-661b-4431-b5bb-648e3d257c32/jobs/1804909?invite=true#step-103-77

I'm not going to touch this here, but whatever we set in 

https://github.com/pytorch/vision/blob/2b16299fdf6d128ab4cc270f28fef0488489f4e4/.circleci/regenerate.py#L284

is completely ignored for Windows.

File: .circleci/config.yml [hunk 2]
```diff
@@ -958,11 +958,6 @@ jobs:
     machine:
       image: ubuntu-2004-cuda-11.4:202110-01
     resource_class: gpu.nvidia.small
-    environment:
-      PYTHON_VERSION: << parameters.python_version >>
-      PYTORCH_VERSION: << parameters.pytorch_version >>
-      UNICODE_ABI: << parameters.unicode_abi >>
-      CU_VERSION: << parameters.cu_version >>
```
Comment (pmeier) [inline .circleci/config.yml:965]: Driveby. We don't need these here, because they are pulled in at

https://github.com/pytorch/vision/blob/2b16299fdf6d128ab4cc270f28fef0488489f4e4/.circleci/config.yml#L957

from

https://github.com/pytorch/vision/blob/2b16299fdf6d128ab4cc270f28fef0488489f4e4/.circleci/config.yml#L222-L226
